utility/unit_of_work/module.go

```diff
@@ -135,9 +135,11 @@ func (uow *baseUtilityUnitOfWork) isProposalBlockSet() bool {
 	return uow.proposalStateHash != "" && uow.proposalProposerAddr != nil
 }
 
-// processTransactionsFromProposalBlock processes the transactions from the proposal block.
-// It also removes the transactions from the mempool if they are also present.
-func (uow *baseUtilityUnitOfWork) processTransactionsFromProposalBlock(txMempool mempool.TXMempool) (err error) {
+// processProposalBlockTransactions processes the transactions from the proposal block stored in the current
+// unit of work. It applies the transactions to the persistence context, indexes them, and removes that from
+// the mempool if they are present.
+func (uow *baseUtilityUnitOfWork) processProposalBlockTransactions(txMempool mempool.TXMempool) (err error) {
+	// CONSIDERATION: should we check that `uow.proposalBlockTxs` is not nil and return an error if so or allow empty blocks?
```

Comment:
IMO We should not allow an entirely empty block instead just push the block time back until we have some Txs to fill it (even partly in the case the network is really not being used at this time).

The only thought is this may affect auto pacemaker mode in Localnet?

Reply:
Fwiw, it's a configuration in Tendermint land.

https://docs.tendermint.com/v0.34/tendermint-core/configuration.html#empty-blocks-vs-no-empty-blocks

![Screenshot 2023-05-19 at 5 45 16 PM](https://github.com/pokt-network/pocket/assets/1892194/4a43a7fb-00e5-41e7-b9ac-28ef85591898)

---

For HotPOKT, right now it's either manual and responsive (at the speed of the network). We have a ticket open to slow it down via a config (#705) which we are saving as a starter task for whoever joins consensus